com.unity.sg2/Editor/GraphDeltaRegistry/FunctionDefinitions/StandardDefinitions/Procedural/HexagonGridNode.cs

```diff
@@ -0,0 +1,186 @@
+    triangleGrid = step(fracHexUVs.xy,fracHexUVs.yx);
+    flooredHexUVs = floor(hexUVs);
+    diamondGrid = fmod(flooredHexUVs.x + flooredHexUVs.y, 3.0);
+    oneDiamonds = step(1.0,diamondGrid);
+    twoDiamonds = step(2.0,diamondGrid);
+    Out.y = dot( triangleGrid, 1.0 - fracHexUVs.yx + (fracHexUVs.x + fracHexUVs.y - 1.0) * oneDiamonds  + (fracHexUVs.yx - 2.0 * fracHexUVs.xy) * twoDiamonds );
+    Out.x = Out.y > LineWidth;
+    Hash_Tchou_2_1_float(floor(flooredHexUVs + oneDiamonds - twoDiamonds * triangleGrid), Out.z);",
+                    new ParameterDescriptor[]
+                    {
+                        new ParameterDescriptor("UV", TYPE.Vec2, Usage.In, REF.UV0),
+                        new ParameterDescriptor("Scale", TYPE.Vec2, Usage.In, new float[] {10.0f, 10.0f}),
+                        new ParameterDescriptor("LineWidth", TYPE.Float, Usage.In, new float[] {0.05f}),
+                        new ParameterDescriptor("Out", TYPE.Vec3, Usage.Out),
+                        new ParameterDescriptor("hexUVs", TYPE.Vec2, Usage.Local),
+                        new ParameterDescriptor("rcpsqrtthree", TYPE.Float, Usage.Local, new float[] {0.57735026919f}),
+                        new ParameterDescriptor("fracHexUVs", TYPE.Vec2, Usage.Local),
+                        new ParameterDescriptor("triangleGrid", TYPE.Vec2, Usage.Local),
+                        new ParameterDescriptor("flooredHexUVs", TYPE.Vec2, Usage.Local),
+                        new ParameterDescriptor("diamondGrid", TYPE.Float, Usage.Local),
+                        new ParameterDescriptor("oneDiamonds", TYPE.Float, Usage.Local),
+                        new ParameterDescriptor("twoDiamonds", TYPE.Float, Usage.Local),
+                    },
+                    new string[]
+                    {
+                        "\"Packages/com.unity.render-pipelines.core/ShaderLibrary/Hashes.hlsl\""
+                    },
+                    isHelper: true
+                ),
+                new(
+                    "One",
+@"   HexGridFunction(UV, Scale, LineWidth, Out);
+    Grid = Out.x;
+    EdgeDistance = Out.y;
+    TileID = Out.z;",
+                    new ParameterDescriptor[]
+                    {
+                        new ParameterDescriptor("UV", TYPE.Vec2, Usage.In, REF.UV0),
+                        new ParameterDescriptor("Scale", TYPE.Vec2, Usage.In, new float[] {10.0f, 10.0f}),
+                        new ParameterDescriptor("LineWidth", TYPE.Float, Usage.In, new float[] {0.05f}),
+                        new ParameterDescriptor("Out", TYPE.Vec3, Usage.Local),
+                        new ParameterDescriptor("Grid", TYPE.Float, Usage.Out),
+                        new ParameterDescriptor("EdgeDistance", TYPE.Float, Usage.Out),
+                        new ParameterDescriptor("TileID", TYPE.Float, Usage.Out),
+                    }
+                ),
+                new(
+                    "Four",
+@"   offset.x = ddx(UV); offset = ddy(UV);
```

Comment:
you have `offset.x =`, is the second offset supposed to be `offset.y`?

Reply:
Copy/paste error.  Thanks!  I fixed it.